# Assembling the _Salmonella enterica_ genome from long read sequences generated by an Oxford Nanopore Sequencer
Iroayo Toki
## Introduction
### Biological Background 
Genome assembly is the reconstruction of the complete genome from segmented DNA sequences generated by a sequencer. This is useful in studying evolution through phylogeny, studying genetic structure of different organisms and can also be used to understand how genes and their products function within an organism (Carpenter & Conlan, 2021). The goal of this report is to understand how to develop a pipeline for the assembly of a bacterial genome (Salmonella enterica) and why specific methods including software and parameters will be chosen after which it will be compared to its reference genome. 
Salmonella enterica is an intracellular pathogen in man and other mammals that causes diseases such as Salmonellosis and Typhoid fever and therefore studying its genome is biologically important in understanding its adaptation and ecological diversity to combat it for the sake of public health (Brown et al., 2021).
One of the major issues that Next generation sequencing techniques faces is repetitive regions, these made it hard for the assemblers to understand where the sequences would fit into, and producing less contiguity in the sequences. This problem is solved by long read sequencing using technologies like the Oxford Nanopore Technology which was used to generate our reads used for this experiment (Logsdon et al., 2020).
To develop the right pipeline for genome assembly the sequencer used and also the type of genetic data have to be taken into account. 

### Methods comparison
## Proposed Methods
### Data description 
### Preprocessing (Quality Control)
### Assembly
### Alignment
### Visualisation
## References
