# Assembling the _Salmonella enterica_ genome from long read sequences generated by an Oxford Nanopore Sequencer
Iroayo Toki

## Introduction

### Biological Background

Genome assembly is the reconstruction of a complete genome from fragmented DNA sequences generated by a sequencing platform. This process is useful for studying evolution through phylogenetic analysis, examining the genetic structure of different organisms, and understanding how genes and their products function within an organism (Carpenter & Conlan, 2021). The goal of this report is to understand how to develop a pipeline for the assembly of a bacterial genome (*Salmonella enterica*), justify the choice of specific methods, software, and parameters, and subsequently compare the resulting assembly to a reference genome.

*Salmonella enterica* is an intracellular pathogen of humans and other mammals that causes diseases such as salmonellosis and typhoid fever. Studying its genome is therefore biologically important for understanding its adaptation and ecological diversity, with direct implications for public health (Brown et al., 2021).

One of the major challenges faced by next-generation sequencing technologies is the presence of repetitive genomic regions. These repeats make it difficult for assemblers to determine the correct placement of sequences, often resulting in fragmented assemblies with reduced contiguity. This limitation can be mitigated through long-read sequencing technologies, such as **Oxford Nanopore sequencing**, which was used to generate the reads for this study (Logsdon et al., 2020).

To develop an effective genome assembly pipeline, both the sequencing technology used and the characteristics of the genetic data must be taken into account.

---

### Methods Comparison

In order to identify appropriate tools for the assembly of long-read bacterial genomes, key factors to consider include **assembly accuracy**, **contiguity**, and **computational efficiency**.

In a 2025 study on *Escherichia coli*, preprocessing steps were shown to significantly affect genome fraction and contiguity. Reads filtered using **Chopper** produced superior assembly results compared to raw reads and reads trimmed using **Porechop**. Although error-corrected reads generated using **Canu** improved accuracy, they occasionally resulted in misassemblies due to overcorrection (Kumar et al., 2025).

The same study also evaluated the runtime of 11 different assembly tools. **Canu** and **NextDenovo** exhibited the longest runtimes, requiring up to four and two hours, respectively, for corrected reads. In contrast, **Flye**, **Miniasm2**, **Hinge**, **Shasta**, and **Raven** completed assemblies in approximately 0.5 hours or less. Miniasm2 was among the fastest assemblers but frequently produced incomplete genomes, whereas Flye and Raven demonstrated a balance between speed and accuracy. Most assemblers produced a single contig, with the exception of Canu, which generated between three and five contigs. Additionally, reads preprocessed using Canu’s error correction step led to fragmented assemblies when used with certain assemblers, such as Flye and Shasta.

In terms of accuracy, **Flye**, **Canu**, and **NECAT** achieved the highest genome fractions, while **Hinge** exhibited the highest number of misassemblies (Kumar et al., 2025).

In a review by Dr. Ryan Wick, **AutoCycler** is described as a command-line tool that combines multiple alternative assemblies of the same genome to generate a higher-quality consensus assembly with fewer sequence and structural errors than any individual long-read assembly. AutoCycler is also capable of producing circularized genomes. It takes as input assemblies generated from subsampled reads and from different assemblers to produce a single high-quality consensus genome (Wick et al., 2025).

These findings suggest that an ideal assembly pipeline would include preprocessing through read filtering using **Chopper**, followed by the combination of top-performing individual assemblers within the **AutoCycler framework**. Computational efficiency can be maintained by selecting assemblers with low to moderate runtimes that also produce accurate and contiguous assemblies.

Finally, a 2024 study on whole-genome alignment demonstrated that **minimap2** outperformed **MUMmer4** and other alignment tools in mapping similar sequences while also exhibiting the fastest runtime (Saada et al., 2024).

---

## Final Methods

### Data Description

The data used for this assembly consisted of raw *Salmonella enterica* reads generated by an **Oxford Nanopore sequencer** using **R10 chemistry** and provided in **FASTQ format**. The dataset is publicly available through the NCBI Sequence Read Archive under the accession **SRR32410565**. The reference genome and annotation were also downloaded from NCBI in **FASTA and GFF format** respectively.

---

### Preprocessing (Quality Control)

Quality control was performed using **Chopper v0.12.0b** to filter low-quality reads. The following parameters were used:

- `--quality` = 10 (minimum mean read quality score)  
- `--minlength` = 1000 (minimum read length)

All other parameters will remain at their default values.
This code used for the filtering step with these parameters is provided below(De Coster & Rademakers, 2023).
```bash
chopper  -q 10 -l 1000  -i SRR32410565.fastq > filtered_seqs.fastq 
```

After filtering

---

### Assembly

Genome assembly will be performed using **AutoCycler v0.5.2** with fully automated commands. Input assemblies will be restricted to:

- **Flye v2.9.6**  
- **Raven v1.8.3**  
- **NECAT v0.0.1_update20200803**

This selection is intended to generate high-quality input assemblies using different assembly paradigms, including **A-Bruijn graph** and **overlap-layout-consensus (OLC)** approaches, in order to reduce errors while maximizing completeness and contiguity. The number of computational threads will be set according to available computing resources (Chen et al., 2021; Kolmogorov et al., 2019; Vaser & Šikić, 2021; Wick et al., 2025).

---

### Alignment

### Alignment

The assembled genome will be aligned to the reference genome using **minimap2 v2.30**. The `-a` option will be specified to output alignments in **SAM format** for downstream processing and visualization. The number of threads will be adjusted based on available computational capacity(Li, 2021).  

Additional options for assembly-to-reference alignment include:

```bash
-x asm5   # assembly to genome alignment
-r2k      # maintain long, unbroken alignments
--cs      # include the exact sequence of every difference in the output
```

---

### Sorting and Indexing

SAM files will be converted to BAM format using:

```bash
samtools view -bS input.sam -o output.bam
```

The resulting BAM files will then be sorted and indexed using:

```bash
samtools sort output.bam -o sorted.bam
samtools index sorted.bam
```
---
### Variant Calling

Variant calling will be performed using **SVIM-asm v1.0.3** (Heller & Vingron, 2021) for assembly-to-reference comparison. The following command will be used:

```bash
svim-asm haploid output_dir sorted.bam reference.fa
```
---
### Visualization

The indexed BAM files will be visualized against the reference genome FASTA file using the **Integrative Genomics Viewer (IGV v2.19.x)** to examine alignment quality and identify sequence variation using the VCF file generated from variant calling (Robinson et al., 2017).


---

### References
Brown, E. W., Bell, R., Zhang, G., Timme, R., Zheng, J., Hammack, T. S., & Allard, M. W. (2021). Salmonella Genomics in Public Health and Food Safety. EcoSal Plus, 9(2). https://doi.org/10.1128/ECOSALPLUS.ESP-0008-2020

Carpenter, S., & Conlan, R. S. (2021). Clinical Functional Genomics. Cancers, 13(18), 4627. https://doi.org/10.3390/CANCERS13184627

Chen, Y., Nie, F., Xie, S. Q., Zheng, Y. F., Dai, Q., Bray, T., Wang, Y. X., Xing, J. F., Huang, Z. J., Wang, D. P., He, L. J., Luo, F., Wang, J. X., Liu, Y. Z., & Xiao, C. Le. (2021). Efficient assembly of nanopore reads via highly accurate and intact error correction. Nature Communications 2021 12:1, 12(1), 60-. https://doi.org/10.1038/s41467-020-20236-7

Danecek, P., Bonfield, J. K., Liddle, J., Marshall, J., Ohan, V., Pollard, M. O., Whitwham, A., Keane, T., McCarthy, S. A., & Davies, R. M. (2021). Twelve years of SAMtools and BCFtools. GigaScience, 10(2). https://doi.org/10.1093/GIGASCIENCE/GIAB008

De Coster, W., & Rademakers, R. (2023). NanoPack2: population-scale evaluation of long-read sequencing data. Bioinformatics, 39(5). https://doi.org/10.1093/BIOINFORMATICS/BTAD311

Heller, D., & Vingron, M. (2021). SVIM-asm: structural variant detection from haploid and diploid genome assemblies. Bioinformatics, 36(22–23), 5519–5521. https://doi.org/10.1093/BIOINFORMATICS/BTAA1034

Kolmogorov, M., Yuan, J., Lin, Y., & Pevzner, P. A. (2019). Assembly of long, error-prone reads using repeat graphs. Nature Biotechnology 2019 37:5, 37(5), 540–546. https://doi.org/10.1038/s41587-019-0072-8

Kumar, M. S., Krishna, M. B., Soman, K. P., Stanley, J., Pourmand, N., Suravajhala, P., & Babu, T. G. S. (2025). Benchmarking long-read assembly tools and preprocessing strategies for bacterial genomes: A case study on E. coli DH5α. Biotechnology Reports, 48, e00931. https://doi.org/10.1016/J.BTRE.2025.E00931

Li, H. (2021). New strategies to improve minimap2 alignment accuracy. Bioinformatics, 37(23), 4572–4574. https://doi.org/10.1093/BIOINFORMATICS/BTAB705

Logsdon, G. A., Vollger, M. R., & Eichler, E. E. (2020). Long-read human genome sequencing and its applications. Nature Reviews. Genetics, 21(10), 597. https://doi.org/10.1038/S41576-020-0236-X

Robinson, J. T., Thorvaldsdóttir, H., Wenger, A. M., Zehir, A., & Mesirov, J. P. (2017). Variant review with the integrative genomics viewer. Cancer Research, 77(21), e31–e34. https://doi.org/10.1158/0008-5472.CAN-17-0337/SUPPLEMENTARY-VIDEO-S1

Saada, B., Zhang, T., Siga, E., Zhang, J., & Magalhães Muniz, M. M. (2024). Whole-Genome Alignment: Methods, Challenges, and Future Directions. Applied Sciences 2024, Vol. 14, Page 4837, 14(11), 4837. https://doi.org/10.3390/APP14114837

Vaser, R., & Šikić, M. (2021). Time- and memory-efficient genome assembly with Raven. Nature Computational Science 2021 1:5, 1(5), 332–336. https://doi.org/10.1038/s43588-021-00073-4

Wick, R. R., Howden, B. P., & Stinear, T. P. (2025). Autocycler: long-read consensus assembly for bacterial genomes. Bioinformatics, 41(9). https://doi.org/10.1093/BIOINFORMATICS/BTAF474
